### ISSUE TITLE
RecursionError after subclassed if sub class defined __setattr__

### ISSUE THREAD

**Issue by xbanke** [seandstewart/typical#62]
* typical version: 2.0.4
* Python version:
* Operating System:

### Description

This might still be related to `MRO`:

```py
import typic


@typic.klass
class Base:
    ...


@typic.klass
class Foo(Base):
    foo: str

    def __setattr__(self, k, v):
        # print('__setattr__')
        super().__setattr__(k, v)


foo = Foo(foo='foo')  # RecursionError
```

**Comment by seandstewart** [comment]
Hey @xbanke - 

Luckily, this isn't directly related to the MRO issues previously.

This is actually an issue I've tried to resolve multiple times while still retaining the high performance of the current implementation. You're in luck, this go (roughly x1000000001) I managed to do so. I'll be pushing a fix and new version shortly.

**Comment by seandstewart** [comment]
This fix has been released in v2.0.5

**Comment by xbanke** [comment]
Thank you very much.
